lib/arel/table.rb

```diff
@@ -96,6 +96,7 @@ def eql? other
     alias :== :eql?
 
     def type_cast_for_database(attribute_name, value)
+      return value if value.is_a? Arel::Nodes::SqlLiteral
```

Comment:
I haven't looked closely at this change (so, e.g., I'm not expressing any opinion on whether this is the right place for the condition), but at a guess, I'd say this should probably be checking for `Arel::Nodes::Node`.